src/Rules/Properties/InvalidCallablePropertyTypeRule.php

```diff
@@ -0,0 +1,120 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Properties;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Node\ClassPropertyNode;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleError;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\ParserNodeTypeToPHPStanType;
+use PHPStan\Type\VerbosityLevel;
+use function sprintf;
+
+/**
+ * @implements Rule<ClassPropertyNode>
+ */
+class InvalidCallablePropertyTypeRule implements Rule
+{
+
+	public function getNodeType(): string
+	{
+		return ClassPropertyNode::class;
+	}
+
+	/**
+	 * @param ClassPropertyNode $node
+	 */
+	public function processNode(Node $node, Scope $scope): array
+	{
+		$nativeType = $node->getNativeType();
+
+		if ($nativeType === null) {
+			return [];
+		}
+
+		if ($this->isExplicitCallableType($nativeType)) {
```

Comment:
This rule can be simplified a lot:

1) Please still keep `if ($nativeType === null) {`.
2) Get `$node->getClassReflection()` and get property reflection via `->getNativeProperty()`.
3) Get `PhpPropertyReflection::getNativeType()`.
4) Use `TypeTraverser::map` to extract CallableType:

```php
TypeTraverser::map($classType, static function (Type $type, callable $traverse) use (&$callableTypes): Type {
	if ($type instanceof UnionType || $type instanceof IntersectionType) {
		return $traverse($type);
	}
	if ($type instanceof CallableType) {
		$callableTypes[] = $type;
	}
	
	return $type;
});
```

We don't need different error message for different situations.